src/Discord/Builders/MessageBuilder.php

```diff
@@ -373,19 +373,25 @@ public function setReplyTo(?Message $message = null): self
      *
      * @param Component $component Component to add.
      *
-     * @throws \InvalidArgumentException Component is not a type of `ActionRow` or `SelectMenu`
-     * @throws \OverflowException        Builder exceeds 5 components.
+     * @throws \InvalidArgumentException Component is not a valid type.
+     * @throws \OverflowException        Builder exceeds component limits.
      *
      * @return $this
      */
     public function addComponent(Component $component): self
     {
-        if (! ($component instanceof ActionRow || $component instanceof SelectMenu)) {
-            throw new \InvalidArgumentException('You can only add action rows and select menus as components to messages. Put your other components inside an action row.');
-        }
+        if ($component instanceof ComponentV2) {
+            if (! ($this->flags & Message::FLAG_V2_COMPONENTS)) {
+                $this->flags |= Message::FLAG_V2_COMPONENTS;
+            }
 
-        if (isset($this->components) && count($this->components) >= 5) {
-            throw new \OverflowException('You can only add 5 components to a message');
+            if (isset($this->components) && count($this->components) >= 10) {
+                throw new \OverflowException('You can only add 10 components to a v2 message');
+            }
+        } elseif (! ($component instanceof Components\ActionRow || $component instanceof Components\SelectMenu)) {
```

Comment:
Please import directly, so it is easy to update/refactor in future (or alias as seen in functions.php)


```
use Discord\Builders\Components\ActionRow;

use Discord\Builders\Components\SelectMenu;
```

```suggestion
        } elseif (! ($component instanceof ActionRow || $component instanceof SelectMenu)) {
```